`skywire-cli visor tp rm --all` panic ; `skywire-cli visor tp rm` also panics (show help menu on no arguments)

```
$ skywire-cli visor tp rm 
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running, locked to thread]:
github.com/skycoin/skywire/cmd/skywire-cli/commands/visor.glob..func38(0xc0005a8a00?, {0x1fca420?, 0x4?, 0x1003b69?})
	/home/runner/work/skywire/skywire/cmd/skywire-cli/commands/visor/transports.go:214 +0x135
github.com/spf13/cobra.(*Command).execute(0x1e5f800, {0x1fca420, 0x0, 0x0})
	/home/runner/work/skywire/skywire/vendor/github.com/spf13/cobra/command.go:944 +0x863
github.com/spf13/cobra.(*Command).ExecuteC(0x1e51aa0)
	/home/runner/work/skywire/skywire/vendor/github.com/spf13/cobra/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/skywire/skywire/vendor/github.com/spf13/cobra/command.go:992
github.com/skycoin/skywire/cmd/skywire-cli/commands.Execute()
	/home/runner/work/skywire/skywire/cmd/skywire-cli/commands/root.go:226 +0x67
main.main()
	/home/runner/work/skywire/skywire/cmd/skywire-cli/skywire-cli.go:9 +0xf

```

same panic when the --all flag is used

```
$ skywire-cli visor tp rm --all
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running, locked to thread]:
github.com/skycoin/skywire/cmd/skywire-cli/commands/visor.glob..func38(0xc00037f300?, {0xc00048a250?, 0x4?, 0x1003b69?})
	/home/runner/work/skywire/skywire/cmd/skywire-cli/commands/visor/transports.go:214 +0x135
github.com/spf13/cobra.(*Command).execute(0x1e5f800, {0xc00048a240, 0x1, 0x1})
	/home/runner/work/skywire/skywire/vendor/github.com/spf13/cobra/command.go:944 +0x863
github.com/spf13/cobra.(*Command).ExecuteC(0x1e51aa0)
	/home/runner/work/skywire/skywire/vendor/github.com/spf13/cobra/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/skywire/skywire/vendor/github.com/spf13/cobra/command.go:992
github.com/skycoin/skywire/cmd/skywire-cli/commands.Execute()
	/home/runner/work/skywire/skywire/cmd/skywire-cli/commands/root.go:226 +0x67
main.main()
	/home/runner/work/skywire/skywire/cmd/skywire-cli/skywire-cli.go:9 +0xf
```

except that in the above instance, using the --all flag, i do have a transport which should have been removed

```
$ skywire-cli visor tp ls
type      id                                       remote_pk                                                              mode        label
sudph     e90fd998-a500-04ef-958d-213144929af2     024223842bba5a31fb0da23dfc822ef783667b74915edbecfc692a9c2a9c4adb56     regular     automatic
```

--- a/cmd/skywire-cli/commands/visor/transports.go
+++ b/cmd/skywire-cli/commands/visor/transports.go
@@ -203,25 +203,20 @@ var rmTpCmd = &cobra.Command{
 	Long:                  "\n    Remove transport(s) by id",
 	DisableFlagsInUseLine: true,
 	Run: func(cmd *cobra.Command, args []string) {
-		//TODO
-		//if removeAll {
-		//	var pks cipher.PubKeys
-		//	internal.Catch(cmd.Flags(), pks.Set(strings.Join(filterPubKeys, ",")))
-		//	tID, err := clirpc.Client(cmd.Flags()).Transports(filterTypes, pks, showLogs)
-		//	internal.Catch(cmd.Flags(), err)
-		//	internal.Catch(cmd.Flags(), clirpc.Client(cmd.Flags()).RemoveTransport(tID))
-		//} else {
-		if args[0] != "" {
-			tpID = args[0]
-		}
-		tID := internal.ParseUUID(cmd.Flags(), "transport-id", tpID)
 		rpcClient, err := clirpc.Client(cmd.Flags())
-		if err != nil {
-			os.Exit(1)
+		if removeAll {
+			internal.Catch(cmd.Flags(), rpcClient.RemoveAllTransports())
+			internal.PrintOutput(cmd.Flags(), "OK", "OK\n")
+		} else if tpID != "" {
+			tID := internal.ParseUUID(cmd.Flags(), "transport-id", tpID)
+			if err != nil {
+				os.Exit(1)
+			}
+			internal.Catch(cmd.Flags(), rpcClient.RemoveTransport(tID))
+			internal.PrintOutput(cmd.Flags(), "OK", "OK\n")
+		} else {
+			internal.PrintOutput(cmd.Flags(), "", cmd.Help())
 		}
-		internal.Catch(cmd.Flags(), rpcClient.RemoveTransport(tID))
-		internal.PrintOutput(cmd.Flags(), "OK", "OK\n")
-		//}
 	},
 }
 

--- a/pkg/transport/manager.go
+++ b/pkg/transport/manager.go
@@ -482,6 +482,22 @@ func (tm *Manager) DeleteTransport(id uuid.UUID) {
 	}
 }
 
+// DeleteAllTransports deregisters all Transports in transport discovery and deletes them locally.
+func (tm *Manager) DeleteAllTransports() {
+	tm.mx.Lock()
+	defer tm.mx.Unlock()
+
+	if tm.isClosing() {
+		return
+	}
+
+	// Deregister transports before closing the underlying transport.
+	for _, tp := range tm.tps {
+		tp.close()
+		delete(tm.tps, tp.Entry.ID)
+	}
+}
+
 // ReadPacket reads data packets from routes.
 func (tm *Manager) ReadPacket() (routing.Packet, error) {
 	p, ok := <-tm.readCh

--- a/pkg/visor/api.go
+++ b/pkg/visor/api.go
@@ -99,6 +99,7 @@ type API interface {
 	Transport(tid uuid.UUID) (*TransportSummary, error)
 	AddTransport(remote cipher.PubKey, tpType string, timeout time.Duration) (*TransportSummary, error)
 	RemoveTransport(tid uuid.UUID) error
+	RemoveAllTransports() error
 	SetPublicAutoconnect(pAc bool) error
 	GetPersistentTransports() ([]transport.PersistentTransports, error)
 	SetPersistentTransports([]transport.PersistentTransports) error
@@ -1189,6 +1190,12 @@ func (v *Visor) AddTransport(remote cipher.PubKey, tpType string, timeout time.D
 // RemoveTransport implements API.
 func (v *Visor) RemoveTransport(tid uuid.UUID) error {
 	v.tpM.DeleteTransport(tid)
+	return nil
+}
+
+// RemoveAllTransports implements API
+func (v *Visor) RemoveAllTransports() error {
+	v.tpM.DeleteAllTransports()
 	return nil
 }
 

--- a/pkg/visor/rpc.go
+++ b/pkg/visor/rpc.go
@@ -527,6 +527,13 @@ func (r *RPC) RemoveTransport(tid *uuid.UUID, _ *struct{}) (err error) {
 	return r.visor.RemoveTransport(*tid)
 }
 
+// RemoveAllTransports removes all Transports from the visor.
+func (r *RPC) RemoveAllTransports(_ *struct{}, _ *struct{}) (err error) {
+	defer rpcutil.LogCall(r.log, "RemoveAllTransports", nil)(nil, &err)
+
+	return r.visor.RemoveAllTransports()
+}
+
 /*
 	<<< AVAILABLE TRANSPORTS >>>
 */

--- a/pkg/visor/rpc_client.go
+++ b/pkg/visor/rpc_client.go
@@ -394,6 +394,11 @@ func (rc *rpcClient) RemoveTransport(tid uuid.UUID) error {
 	return rc.Call("RemoveTransport", &tid, &struct{}{})
 }
 
+// RemoveAllTransports calls RemoveAllTransports.
+func (rc *rpcClient) RemoveAllTransports() error {
+	return rc.Call("RemoveAllTransports", &struct{}{}, &struct{}{})
+}
+
 func (rc *rpcClient) DiscoverTransportsByPK(pk cipher.PubKey) ([]*transport.Entry, error) {
 	entries := make([]*transport.Entry, 0)
 	err := rc.Call("DiscoverTransportsByPK", &pk, &entries)
@@ -1155,6 +1160,14 @@ func (mc *mockRPCClient) RemoveTransport(tid uuid.UUID) error {
 			}
 		}
 		return fmt.Errorf("transport of id '%s' is not found", tid)
+	})
+}
+
+// RemoveAllTransports implements API.
+func (mc *mockRPCClient) RemoveAllTransports() error {
+	return mc.do(true, func() error {
+		mc.o.Transports = []*TransportSummary{}
+		return nil
 	})
 }
 

